internal/kibana/packages.go

```diff
@@ -27,6 +30,47 @@ func (c *Client) InstallPackage(ctx context.Context, name, version string) ([]pa
 	return processResults("install", statusCode, respBody)
 }
 
+// EnsureZipPackageCanBeInstalled checks whether or not it can be installed a package using the upload API.
+func (c *Client) EnsureZipPackageCanBeInstalled(ctx context.Context) error {
+	path := fmt.Sprintf("%s/epm/packages", FleetAPI)
+
+	req, err := c.newRequest(ctx, http.MethodPost, path, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/zip")
+	req.Header.Add("elastic-api-version", "2023-10-31")
+
+	statusCode, respBody, err := c.doRequest(req)
+	if err != nil {
+		return fmt.Errorf("could not install zip package: %w", err)
+	}
+	if statusCode == http.StatusOK {
+		return nil
+	}
+
+	var resp struct {
+		StatusCode int    `json:"statusCode"`
+		Error      string `json:"error"`
+		Message    string `json:"message"`
+	}
+
+	if err := json.Unmarshal(respBody, &resp); err != nil {
+		return fmt.Errorf("could not unmarhsall response to JSON: %w", err)
+	}
+
+	// If the stack allows to use the upload API, the response is like this one:
+	// {
+	//   "statusCode":400,
+	//   "error":"Bad Request",
+	//   "message":"Error during extraction of package: Error: end of central directory record signature not found. Assumed content type was application/zip, check if this matches the archive type."
+	// }
+	if resp.Error == "Forbidden" && resp.Message == "Requires Enterprise license" {
+		return ErrNotSupported
+	}
```

Comment:
This request should never return `StatusOK`, right? Should we handle unexpected statuses more explicitly?

```suggestion
	switch statusCode {
	case http.StatusBadRequest:
		// If the stack allows to use the upload API, the response is like this one:
		// {
		//   "statusCode":400,
		//   "error":"Bad Request",
		//   "message":"Error during extraction of package: Error: end of central directory record signature not found. Assumed content type was application/zip, check if this matches the archive type."
		// }
		return nil
	case http.StatusForbidden:
		var resp struct {
			Message    string `json:"message"`
		}
		if err := json.Unmarshal(respBody, &resp); err != nil {
			return fmt.Errorf("could not unmarhsall response to JSON: %w", err)
		}
		if resp.Message == "Requires Enterprise license" {
			return ErrNotSupported
		}
	}

	return fmt.Errorf("unexpected response (status code %d): %s", statusCode, string(respBody))
```

Reply:
> This request should never return StatusOK, right? Should we handle unexpected statuses more explicitly?

That's right, that request should just return Bad Request or Forbidden. Other status, like `StatusOK` would not be possible.

I'll apply your suggestion, thanks!

Reply:
Applied suggestion here https://github.com/elastic/elastic-package/pull/2511/commits/538149c0e2123d7d0176ce03c1fc864974d5bff9